### DISCUSSION TITLE
How to get selected messages in the active chat

### DISCUSSION THREAD

**Discussion by Birol2010** [wppconnect-team/wa-js#2997] [Q&A]
I searched documents but not found the selected messages in activechat object.
I also wonder how to select the messages.

**Comment by icleitoncosta** [comment]
> I searched documents but not found the selected messages in activechat object.
> I also wonder how to select the messages.


The requested functions don't exist in WAJS, but I added the label for us to add them in the future.

**Comment by janardhan5031** [comment]
> I searched documents but not found the selected messages in activechat object. I also wonder how to select the messages.

Me also trying to develop this feature. let us know if any body develop this

**Comment by ariona** [comment]
Any update on this issue?

**Comment by ariona** [comment]
My current solution is by traversing DOM tree,

```js
function getSelectedMessages() {
  let selectedIds = Array.from(document.querySelectorAll("[role='row'] [data-id]")).filter(i=>i.querySelector('input:checked')).map(i=>i.dataset["id"])
  const selectedMsgs = WPP.whatsapp.MsgStore._models.filter(i=>selectedIds.includes(i.id._serialized))

  return selectedMsgs
}

// will return selected Messages object, and empty array if no selected message
console.log(getSelectedMessages())
```

It would be great to have this function baked into the lib.

**Comment by bracketsApps007** [comment]
any better option for this ?
because above metod will not work for 2+ stickers
send 2 or + stickers to someone and select them one by one it is returning 'groupd-seickers--XXXXXX' as id for all stickers

**Comment by manfe** [comment]
I've converted the original issue, to this discussion thread to keep in right place. If you guys don't need any  more info here, let's close it.

But in my 2 cents here, these faced problems are not related directly to wa-js, it's more how to handle DOM manipuilation. So I would close this thread in future.
